packages/core/src/interpreter.ts

```diff
@@ -91,22 +66,29 @@ const defaultOptions: InterpreterOptions = {
   devTools: false
 };
 
+type InternalStateFrom<
+  TBehavior extends Behavior<any, any, any>
+> = TBehavior extends Behavior<infer _, infer __, infer TInternalState>
+  ? TInternalState
+  : never;
+
 export class Interpreter<
-  TContext extends MachineContext,
-  TEvent extends EventObject = EventObject,
-  TResolvedTypesMeta = TypegenDisabled
-> implements ActorRef<TEvent, State<TContext, TEvent, TResolvedTypesMeta>> {
+  TBehavior extends Behavior<any, any>,
+  TEvent extends EventObject = EventFromBehavior<TBehavior>
+> implements ActorRef<TEvent, SnapshotFrom<TBehavior>> {
```

Comment:
This type param list is incorrect as it allows such things:
```ts
new Interpreter<Behavior<{ type: 'BAR' }>, { type: 'FOO' }>({} as any)
```

It should either only use `TBehavior` or it should use all of the `Behavior`'s generic slots. I think that at the end of the day I prefer the latter as it's more common and doesn't suffer from problems with `never`.

Reply:
I'd prefer to consolidate generics, so I'll play around with having one `TBehavior` unless you want to.

Reply:
Note that separate generics are somewhat better as they can be "forwarded" in generic contexts:
https://twitter.com/TitianCernicova/status/1501237710803156992

I think that this place here is also a little bit different than the `createMachine` thing where we want to supply a huge list of generics explicitly. Here, we can rather easily rely on inference.